cmd/runner/rabbit.go

```diff
@@ -28,21 +29,41 @@ var (
 // retrieving and handling StudioML workloads within a self hosted
 // queue context
 
-func initRMQ() (rmq *runner.RabbitMQ) {
-	// NewRabbitMQ takes a URL that has no credentials or tokens attached as the
-	// first parameter and the user name password as the second parameter
-	creds := ""
-	qURL, errGo := url.Parse(os.ExpandEnv(*amqpURL))
-	if errGo != nil {
-		logger.Warn(kv.Wrap(errGo).With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+func rmqConfig() (creds string, queueURL string, mgtURL string, err kv.Error) {
+	if len(*amqpURL) != 0 {
+		qURL, errGo := url.Parse(os.ExpandEnv(*amqpURL))
+		if errGo != nil {
+			logger.Warn(kv.Wrap(errGo).With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
+		if qURL.User != nil {
+			creds = qURL.User.String()
+		} else {
+			logger.Warn(kv.NewError("missing credentials in url").With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
+		qURL.User = nil
+		queueURL = qURL.String()
 	}
-	if qURL.User != nil {
-		creds = qURL.User.String()
-	} else {
-		logger.Warn(kv.NewError("missing credentials in url").With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+
+	if len(*amqpMgtURL) != 0 {
+		qMgtURL, errGo := url.Parse(os.ExpandEnv(*amqpMgtURL))
+		if errGo != nil {
+			logger.Warn(kv.Wrap(errGo).With("url", *amqpMgtURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
```

Comment:
If Parse() call goes bad, are we always able to continue after this warning?

Reply:
Yes I will with an empty result.  This then bails out later on when the connection is attempted, or is ignored.